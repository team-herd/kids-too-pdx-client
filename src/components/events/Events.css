.Event {
  display: flex;
  justify-content: center;
  ul {
    width:85vw;
    list-style: none;
    padding: 0;
    li {
      padding: 1em;
      border: 1px solid black;
      border-radius: 10px;
      background-color: white;
      margin-bottom: 0.8em;
    }
  }
}
@media only screen and (min-width: 1000px) {
  .Event {
    ul {
      display: grid;
      grid-template-columns: repeat(auto-fill, minmax(20vw, 1fr));
      margin-top: 15em;
    }
    li {
      margin: 1em;
    }
  }
}