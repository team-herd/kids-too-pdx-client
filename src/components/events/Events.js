import React from 'react';
import Event from './Event';
import PropTypes from 'prop-types';

function Events({ events }) {
  const listOfEvents = events.map((event, i) => {
    return <li key={i}><Event event={event}/></li>;
  });
  return (
    <ul>
      {listOfEvents}
    </ul>
  );
}
Events.propTypes = {
  events: PropTypes.array
};
export default Events;
