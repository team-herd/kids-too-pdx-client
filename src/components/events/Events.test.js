import React from 'react';
import renderer from 'react-test-renderer';
import Events from './Events';
import { MemoryRouter } from 'react-router-dom';

describe('Events', () => {
  it('matches snapshot for Events', () => {
    const props = {
      events: [
        {
          _id: 1,
          image: 'image.com',
          time: '2pm',
          name: 'House Event',
          date: new Date().toString(),
          location: {
            venue: 'My House'
          },
          price: 250,
          ageMin: 9,
          ageMax: 12,
          category: 'Art',
          description: 'The bestest event'
        },
        {
          _id: 2,
          image: 'image.com',
          time: '2pm',
          name: 'House Event',
          date: new Date().toString(),
          location: {
            venue: 'My House'
          },
          price: 150,
          ageMin: 9,
          ageMax: 12,
          category: 'Art',
          description: 'The bestest event'
        },
        {
          _id: 3,
          image: 'image.com',
          time: '2pm',
          name: 'House Event',
          date: new Date().toString(),
          location: {
            venue: 'My House'
          },
          price: 50,
          ageMin: 9,
          ageMax: 12,
          category: 'Art',
          description: 'The bestest event',
          details: true
        }
      ],

      onChange: jest.fn(),

      onSubmit: jest.fn()
    };

    const tree = renderer.create(
      <MemoryRouter>
        <Events {...props} />
      </MemoryRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
