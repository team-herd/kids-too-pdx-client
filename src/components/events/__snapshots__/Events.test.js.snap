// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events matches snapshot for Events 1`] = `
Array [
  <form
    className="Select"
    onSubmit={[MockFunction]}
  >
    <select
      name="category"
      onChange={[MockFunction]}
    >
      <option
        hidden={true}
        value="DEFAULT"
      >
        Category
      </option>
      <option
        value="Sports"
      >
        Sports
      </option>
      <option
        value="Art"
      >
        Art
      </option>
      <option
        value="Music"
      >
        Music
      </option>
      <option
        value="Performance"
      >
        Performance
      </option>
      <option
        value="Educational"
      >
        Educational
      </option>
      <option
        value="Volunteer"
      >
        Volunteer
      </option>
    </select>
    <label>
      Price:
      <input
        name="price"
        onChange={[MockFunction]}
        placeholder="max"
        type="number"
      />
    </label>
    <label>
      Age:
      <input
        name="ageMin"
        onChange={[MockFunction]}
        placeholder="min age"
        type="number"
      />
      <input
        name="ageMax"
        onChange={[MockFunction]}
        placeholder="max age"
        type="number"
      />
    </label>
    <button
      type="submit"
    >
      Search
    </button>
  </form>,
  <section
    className="Event"
  >
    <ul>
      <li>
        <section
          className="Event"
        >
          <img
            alt="event picture"
            src="image.com"
          />
          <a
            href="/events/1"
            onClick={[Function]}
          >
            <h2>
              House Event
            </h2>
          </a>
          <p>
            <span>
              Date:
            </span>
            Fri Mar 29 2019 08:53:35 GMT-0700 (Pacific Daylight Time)
          </p>
          <p>
            <span>
              Time:
            </span>
             
            2pm
          </p>
          <p>
            <span>
              Location:
            </span>
            My House
          </p>
          <p>
            <span>
              Cost:
            </span>
            250
          </p>
          <p>
            <span>
              Age:
            </span>
             
            9-12
          </p>
          <p>
            Art
          </p>
        </section>
      </li>
      <li>
        <section
          className="Event"
        >
          <img
            alt="event picture"
            src="image.com"
          />
          <a
            href="/events/2"
            onClick={[Function]}
          >
            <h2>
              House Event
            </h2>
          </a>
          <p>
            <span>
              Date:
            </span>
            Fri Mar 29 2019 08:53:35 GMT-0700 (Pacific Daylight Time)
          </p>
          <p>
            <span>
              Time:
            </span>
             
            2pm
          </p>
          <p>
            <span>
              Location:
            </span>
            My House
          </p>
          <p>
            <span>
              Cost:
            </span>
            150
          </p>
          <p>
            <span>
              Age:
            </span>
             
            9-12
          </p>
          <p>
            Art
          </p>
        </section>
      </li>
      <li>
        <section
          className="Event"
        >
          <img
            alt="event picture"
            src="image.com"
          />
          <a
            href="/events/3"
            onClick={[Function]}
          >
            <h2>
              House Event
            </h2>
          </a>
          <p>
            <span>
              Date:
            </span>
            Fri Mar 29 2019 08:53:35 GMT-0700 (Pacific Daylight Time)
          </p>
          <p>
            <span>
              Time:
            </span>
             
            2pm
          </p>
          <p>
            <span>
              Location:
            </span>
            My House
          </p>
          <p>
            <span>
              Cost:
            </span>
            50
          </p>
          <p>
            <span>
              Age:
            </span>
             
            9-12
          </p>
          <p>
            Art
          </p>
        </section>
      </li>
    </ul>
  </section>,
]
`;
